src/context/AppProvider/hooks/useAccountsFetchQuery.tsx

```diff
@@ -14,27 +13,15 @@ import { walletSupportsChain } from '@/hooks/useWalletSupportsChain/useWalletSup
 import { deriveAccountIdsAndMetadata } from '@/lib/account/account'
 import { isUtxoChainId } from '@/lib/utils/utxo'
 import { portfolio, portfolioApi } from '@/state/slices/portfolioSlice/portfolioSlice'
-import { selectEnabledWalletAccountIds } from '@/state/slices/selectors'
-import { useAppDispatch, useAppSelector } from '@/state/store'
+import { store, useAppDispatch, useAppSelector } from '@/state/store'
 
 export const useAccountsFetchQuery = () => {
   const dispatch = useAppDispatch()
-  const enabledWalletAccountIds = useAppSelector(selectEnabledWalletAccountIds)
   const { supportedChains } = usePlugins()
   const { isSnapInstalled } = useIsSnapInstalled()
-  const previousIsSnapInstalled = usePrevious(isSnapInstalled)
   const { deviceId, wallet } = useWallet().state
-
-  const hasManagedAccounts = useMemo(() => {
-    // MM without snap doesn't allow account management - if the user just installed the snap, we
-    // know they don't have managed accounts. NOTE - the values we're comparing here are
-    // `boolean | null`, so explicit comparison is needed!
-    if (previousIsSnapInstalled === false && isSnapInstalled === true) {
-      return false
-    }
-    // We know snap wasn't just installed in this render - so if there are any requestedAccountIds, we assume the user has managed accounts
-    return enabledWalletAccountIds.length > 0
-  }, [isSnapInstalled, previousIsSnapInstalled, enabledWalletAccountIds.length])
```

Comment:
q: have we tested manual account management bits / snap just installed flow?
This should also be part of the risks/testing steps btw

Reply:
it does feel like there is an edge case here and we will want to react on the state of snap installed to retrigger the query.

Reply:
Yep, added to the testing steps! Confirming it works as intended, I think @kaladinlight did give it a try too

Reply:
Looks like this breaks MM snaps disconnect and accounts management rehydration. https://jam.dev/c/06b984ea-e74c-4210-87ea-f173784c603a